src/connectors/hits-per-page/connectHitsPerPage.ts

```diff
@@ -235,11 +220,20 @@ You may want to add another entry to the \`items\` option with this value.`
         };
       },
 
-      getWidgetRenderState({ state, results }) {
+      getWidgetRenderState({ state, results, createURL }) {
+        connectorState.createURLFactory = helperState => value => {
+          return createURL(
+            helperState.setQueryParameter(
+              'hitsPerPage',
+              !value && value !== 0 ? undefined : value
+            )
+          );
+        };
+
         return {
           items: transformItems(normalizeItems(state)),
           refine: connectorState.setHitsPerPage,
-          createURL: connectorState.createURLFactory,
+          createURL: connectorState.createURLFactory(state),
           hasNoResults: results?.nbHits === 0,
```

Comment:
this should be `results ? results.nbHits === 0 : true`, since otherwise it's wrong in init :)